compiler-core/src/analyse.rs

```diff
@@ -342,33 +345,40 @@ pub fn register_import(
         }
     }
 
-    if unqualified.is_empty() {
-        // When the module has no unqualified imports, we track its usage
-        // so we can warn if not used by the end of the type checking
+    if !discarded {
```

Comment:
This is going to change due to the removal of the`discarded` field, but generally let's use guard clauses rather than if with indentation.

Indentation is hard to read because you have to read the conditional and then scan down to the bottom of the function to see if there's an `else`, while with a guard you don't have to hold this unresolved question while reading the rest.